### PR TITLE
S3 LQ model

### PR DESCRIPTION
**Pull request by wkotlarski**
For some project I had to create a model with a $S_3$ leptoquark (it's the one with $(\bar{3}, 3, 1/3)$ $SU(3)_c \times SU(2)_L \times U(1)_Y$ quantum numbers). It's defined in Appendix A [here](https://inspirehep.net/literature/1805490). SARAH does not have it. If you think it might be useful to include it, we can discuss the details of the setup.

### CODE REVIEW

File: sarah/S3/S3.m [hunk 1]
```diff
@@ -0,0 +1,107 @@
+Off[General::spell]
+
+Model`Name = "S3";
+Model`NameLaTeX ="S3 Leptoquark";
+Model`Authors = "Daniele Rizzo, Abhishek Chikkaballi Ramalingegowda and Wojciech Kotlarski";
+Model`Date = "2023-04-17";
+
+(*-------------------------------------------*)
+(*Starting comment from SM implementation*)
```
Comment (Expander) [inline sarah/S3/S3.m:9]: Are these comments correct? (I thought it is a new model file that you've wrote. :) )

File: model_files/S3/FlexibleSUSY.m.in [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+   {FlexibleSUSYLowEnergy, {
+         {20, FlexibleSUSYObservable`AMM[Fe[1]]},
+         {21, FlexibleSUSYObservable`AMM[Fe[2]]},
+         {26, FlexibleSUSYObservable`BrLToLGamma[Fe[2] -> {Fe[1], VP}]},
+         {27, FlexibleSUSYObservable`bsgamma}
+      }
+   }
+};
+
+SARAH`ParameterDefinitions = Append[
```
Comment (Expander) [inline model_files/S3/FlexibleSUSY.m.in:61]: I think this can be removed, because the model file seems to contain this description.

**Comment by Expander** [comment]
Just to be sure: Did you run `CheckModel[]` to see if SARAH prints any warnings?

File: model_files/S3/FlexibleSUSY.m.in (hunk 2 above)
Comment (wkotlarski) [reply model_files/S3/FlexibleSUSY.m.in]: Removed in 60049358013daf90eece343a9cb25ef2d6449b93.

File: sarah/S3/S3.m (hunk 1 above)
Comment (Expander) [reply sarah/S3/S3.m:9]: Removing them would be good, in my opinion, because I think the SM history does not apply here / does not make sense here.

**Comment by Expander** [comment]
One question: Does the default SLHA input file yield a (more or less) physically plausible output?

**Comment by pathron** [comment]
Thanks for adding this!  Just looking I don't see the scalar leptoquark models Douglas wrote in the development repo. I guess these were never added?  There is also a mixed  S1 S3 model from my other MW paper. Bin Zhu mostly wrote the sarah files (i think i just fixed some minor things) so I'd need to ask him if we put them in FS or SARAH but should we try to add all these?

**Comment by wkotlarski** [comment]
> Thanks for adding this! Just looking I don't see the scalar leptoquark models Douglas wrote in the development repo. I guess these were never added? There is also a mixed S1 S3 model from my other MW paper. Bin Zhu mostly wrote the sarah files (i think i just fixed some minor things) so I'd need to ask him if we put them in FS or SARAH but should we try to add all these?

The only ones I've found are from the 750 GeV paper but there is no $S_3$ one there.

**Comment by wkotlarski** [comment]
> One question: Does the default SLHA input file yield a (more or less) physically plausible output?

If we set the input to
```
Block YS3LIN
    1   1   0     # YS3LInput(1,1)
    1   2   0     # YS3LInput(1,2)
    1   3   0     # YS3LInput(1,3)
    2   1   0     # YS3LInput(2,1)
    2   2   0     # YS3LInput(2,2)
    2   3   0.1   # YS3LInput(2,3)
    3   1   0.01  # YS3LInput(3,1)
    3   2   0.1   # YS3LInput(3,2)
    3   3   0.1   # YS3LInput(3,3)
```
we'll get
```
Block FlexibleSUSYLowEnergy Q= 1.73000000E+02
    20    -2.20137041E-19   # Delta(g-2)/2 of Fe(1) (calculated with FlexibleSUSY)
    21    -1.00128606E-12   # Delta(g-2)/2 of Fe(2) (calculated with FlexibleSUSY)
    26     2.35492123E-12   # BR(Fe1 -> Fe0 VP)
    27     2.20382799E-03   # Re(C7) for b -> s gamma
```
which seems reasonanble. I cannot check $b \to s \gamma$ since after the last update `flavio` doesn't work.

Side question: since this model does generate $b \to s \gamma$, should we include CKM matrix? Or should we split it into 2 models: `S3` and `S3CKM`?

File: sarah/S3/S3.m (hunk 1 above)
Comment (wkotlarski) [reply sarah/S3/S3.m]: Tweaked in 9cf2b46a8b9461d34ba5c96c0f2b661e9e0ec5e3.

**Comment by Expander** [comment]
I think adding the CKM matrix directly would be best.

**Comment by Expander** [comment]
Actually, I've got one secret wish: Could we improve the indentation if `parameters.m` and `particles.m` and remove trailing whitespace etc.? :)

**Comment by wkotlarski** [comment]
BTW, why don't we have a `SMCKM`? I wanted to copy stuff from it but apparently we only have `CMSSMCKM` and `MRSSM2CKM`.

**Comment by Expander** [comment]
> BTW, why don't we have a `SMCKM`? I wanted to copy stuff from it but apparently we only have `CMSSMCKM` and `MRSSM2CKM`.

I think we never had an application for it. :D

**Comment by wkotlarski** [comment]
I've added the CKM matrix in ed7539a7ada6ff803a40a419daf9eac553f47f9a.

**Comment by Expander** [comment]
> > Actually, I've got one secret wish: Could we improve the indentation if `parameters.m` and `particles.m` and remove trailing whitespace etc.? :)
> 
> I would not re-indent entire files because, again, those are mostly copies of `SM`'s `parameters.m` and `particles.m`. If we'd do that, it would be hard to see what was change. I would only indent parts we've added.

I was afraid you'd say that. ;)

File: sarah/S3/particles.m [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                 Mass -> {0},
+                 LaTeX -> {"H^+","H^-"},
+                 OutputName -> {"Hp","Hm"},
+                 ElectricCharge->1
+                 }},  
+
+      {phi13,  { Description -> "Phi 1/3",
+                 PDG -> {100},
+                 LaTeX -> {"\\phi_{1/3}","\\phi_{1/3}"},
+                 OutputName -> "phi1",
```
Comment (Expander) [inline sarah/S3/particles.m:78]: I'd propose to change this to `phi13`. (Similarly to the other phi fields.)

File: sarah/S3/particles.m [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                 ElectricCharge->1/3
+                 }},
+
+      {phi43,  { Description -> "Charged phi43",
+                 PDG -> {101},
+                 LaTeX -> {"\\phi_{4/3}","\\phi_{-4/3}"},
+                 OutputName -> {"phi2","conjphi2"},
```
Comment (Expander) [inline sarah/S3/particles.m:85]: Here maybe `phi43`

Comment (wkotlarski) [reply sarah/S3/particles.m]: Done in 416c40c727b1fff32493afb31ace937f455320d0.

**Comment by wkotlarski** [comment]
@Expander, @pathron, can this be merged now?

**Comment by Expander** [comment]
Yes, please!

**Comment by pathron** [comment]
Sorry looks like you already merged this, thanks!

**Pull request merged**